### PR TITLE
Change ModuleGroupMap to ModuleSet

### LINKED ISSUE #121: Change ModuleGroupMap to a Set
Module's already have a group system. Instead of making a mapping for a person (hashmap), I should make a set of modules, which can only contain one group at a time. 

This change should not affect other components in the project, and only a simple refactor of the underlying working of ModuleGroupMap

### PR DESCRIPTION
**Pull request by FergusMok**
Fixes #121, #116, #191, #188
This will allow a person's module to have multiple groups (Labs, Tutorials, Lecture etc..) instead of only having 1.

### CODE REVIEW

File: src/main/java/seedu/edrecord/model/module/ModuleSet.java [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+     * containsModule should be called together with this method.
+     * @return boolean to show if the set contains the given group in given module.
+     */
+    public boolean containsGroupInModule(Module mod, Group grp) {
+        for (Module module : this.getModules()) {
+            if (module.getCode().equals(mod.getCode())) {
+                if (module.hasGroup(grp)) {
+                    return true;
+                }
+            }
```
Comment (chaiwanlin) [inline src/main/java/seedu/edrecord/model/module/ModuleSet.java:110]:
```suggestion
            if (module.getCode().equals(mod.getCode()) && module.hasGroup(grp)) {
                return true;
            }
```
Less arrow-head style code

**Review by chaiwanlin — CHANGES_REQUESTED** [review]
Just a small suggestion, otherwise lgtm

**Review by marcusc55 — APPROVED** [review]
LGTM

**Review by chaiwanlin — APPROVED** [review]
lgtm

**Pull request merged**